pkg/controller/disruption/disruption.go

```diff
@@ -321,17 +320,24 @@ func (dc *DisruptionController) getScaleController(ctx context.Context, controll
 	return &controllerAndScale{scale.UID, scale.Spec.Replicas}, nil
 }
 
-func (dc *DisruptionController) implementsScale(gv schema.GroupVersion, kind string) (bool, error) {
-	resourceList, err := dc.discoveryClient.ServerResourcesForGroupVersion(gv.String())
+func (dc *DisruptionController) implementsScale(gvr schema.GroupVersionResource) (bool, error) {
+	resourceList, err := dc.discoveryClient.ServerResourcesForGroupVersion(gvr.GroupVersion().String())
 	if err != nil {
 		return false, err
 	}
+
+	scaleSubresourceName := fmt.Sprintf("%s/scale", gvr.Resource)
 	for _, resource := range resourceList.APIResources {
-		if resource.Kind != kind {
+		if resource.Name != scaleSubresourceName {
 			continue
 		}
-		if strings.HasSuffix(resource.Name, "/scale") {
-			return true, nil
+
+		for _, scaleGv := range scaleclient.NewScaleConverter().ScaleVersions() {
```

Comment:
add a unit test case that exercises a `.../scale` subresource that is not one of the recognized Scale types?